tests/test_throttler.py

```diff
@@ -79,3 +83,30 @@ def test_flush() -> None:
     f1.flush()
     time.sleep(0.2)
     mock1.assert_called_once()
+
+
+@pytest.mark.parametrize("deco", [debounced, throttled])
+def test_throttled_debounced_signature(deco: Callable) -> None:
+    mock = Mock()
+
+    @deco(timeout=0)
+    def f1(x: int) -> None:
+        """Doc."""
+        mock(x)
+
+    # make sure we can still inspect the signature
+    assert signature(f1).parameters["x"] == Parameter(
+        "x", Parameter.POSITIONAL_OR_KEYWORD, annotation=int
+    )
+
+    # make sure these are connectable
+    sig = SignalInstance((int, int, int))
+    sig.connect(f1)
+    sig.emit(1, 2, 3)
+    time.sleep(0.1)
```

Comment:
Always sad to see hopeful sleeps in tests, but I guess there's no equivalent of `pytest.waitSignal` in psygnal? Doesn't block this PR, as there are other tests with similar. Let me know if you'd be interested in having something like `waitSignal` and I could attempt a crude implementation.

Reply:
> Always sad to see hopeful sleeps in tests.

yeah, I totally agree.
There's no equivalent (however... `pytest.waitSignal` *does* work just fine with psygnal signals too... since it's got the same API as qt)... I could use that here too, just didn't want to have to bring qt into this :joy:

Reply:
fwiw, I did test a huge range of sleeps (all the way down to 0) and it continued to work.  so I could probably remove it completely.  mostly it's there to induce a beat, if nothing more

Reply:
was able to remove the sleep here